### PR TITLE
Include default values in runconfig written to HDF5 metadata

### PR DESCRIPTION
**Pull request by LiangJYu**
This PR adds default values to runconfig string written to the HDF5 metadata. Currently only the user input runconfig is written to HDF5 metadata.

### CODE REVIEW

**Review by vbrancat — COMMENTED** [review]
LGTM. I do not have any comment on the code. I have not had time to test it. Will do this after finishing some processing on aurora. Will approve after that

File: src/compass/utils/runconfig.py [hunk 1]
```diff
@@ -453,5 +458,5 @@ def to_yaml(self):
         '''Dump runconfig as string to sys.stdout
         '''
         self_as_dict = self.as_dict()
-        yaml = YAML(typ='safe')
-        yaml.dump(self_as_dict, sys.stdout)
+        yaml_obj = YAML(typ='safe')
```
Comment (scottstanie) [inline src/compass/utils/runconfig.py:461]: just checking- any reason for this change?

File: src/compass/utils/runconfig.py [hunk 2]
```diff
@@ -346,15 +352,14 @@ def load_from_yaml(cls, yaml_path: str, workflow_name: str) -> RunConfig:
                 sns.input_file_group.reference_burst.file_path,
                 sns.input_file_group.burst_id)
 
-        # For saving entire file as string to metadata. Stop gap for writing
-        # dict to individual elements to HDF5 metadata
-        with open(yaml_path, 'r') as f_yaml:
-            entire_yaml = f_yaml.read()
+        # For saving entire file with default fill-in as string to metadata.
```
Comment (scottstanie) [inline src/compass/utils/runconfig.py:355]:
```suggestion
        # For saving entire file with defaults filled-in as string to metadata.
```

**Review by scottstanie — APPROVED** [review]
I also haven't ran the test yet, but it looks like it should be fine. 
I just wanted to check, since it wasn't obvious to me at the first read
Is the only place where the `cfg` is mutated in the `load_validate_yaml`? 
- if so, maybe we should move the `user_plus_default_yaml_str = yaml.dump(cfg)` right below to indicate that the changing of `cfg` is set
- if not, could we put some comma/indicate in the name somehow that further changes/mutations are done on the cfg? It's just hard to tell from the function where the changes might be happening

File: src/compass/utils/runconfig.py (hunk 1 above)
Comment (LiangJYu) [reply src/compass/utils/runconfig.py:461]: Not necessary, but appended `_obj` to avoid conflict with `import yaml` up top.

**Pull request merged**